app/src/main/AndroidManifest.xml

```diff
@@ -82,6 +83,13 @@
                 android:resource="@xml/file_paths" />
 
         </provider>
+
+        <!-- Enables VPN auto-start on API23 devices -->
+        <receiver android:name="org.mozilla.firefox.vpn.receivers.LegacyBootBroadcastReceiver">
```

Comment:
We could consider using the `enabled` flag and setting it by a resource that is different depending on the API level (`values-v*`). But your runtime check is good too, I think.

Reply:
Ah, neat. I wasn't quite sure how to do that without googling a bunch, so runtime check it is :)